### ISSUE TITLE
Why is the matrix in RNN designed this way？

### ISSUE THREAD

**Issue by ZincJT** [sahapriyabrata/PhICNet#1]
Wcc，wcu，whc。Why are these three matrices designed this way？I have read your paper, but I don't understand these three matrices very well。As well as，What does sf mean? Why source mapV(t)/sf?My English is not very good, if the author does not understand my question, I will edit the question after reading the reply, I really appreciate your answer to my doubts。

**Comment by sahapriyabrata** [comment]
Consider the case when the temporal order is $n=2$. In this case, we need the past two samples to compute the next one.
Therefore, the current cell memory, say at time $t$, is $C_t = [U_t, U_{t-1}]$ (see equation 6). Similarly, cell memory at time $(t-1)$ was $C_{t-1} = [U_{t-1}, U_{t-2}]$. Now, to get $C_t$ from its previous version $C_{t-1}$ and current state map $U_t$, we can write this:
$C_t = [0, \  0; 1, \  0] C_{t-1} + [1, \ 0] U_t = W_{cc} C_{t-1} + w_{cu}  U_t$, which is equation 7. 

$w_{hc}$ is determined from the finite difference approximation of $\frac{d^2 u}{d t^2} = \frac{u_{t+1} - 2 u_{t} + u_{t-1}}{\Delta t^2} =f$. So $U_{t+1}$ can be obtained as $U_{t+1} = 2 U_{t} - U_{t-1} + \Delta t^2 f = [2, -1] C_t + \Delta t^2 f = w_{hc} C_t + \Delta t^2 f$.

Hope this helps.

**Comment by ZincJT** [comment]
I really appreciate you being able to answer my question and I also understand your reply.But I read your code carefully. Wcc is a matrix with zero elements, and it doesn't change，The same is true of the other matrices. I feel a little different here than in your paper.I also tried to test with my own data sets and found that well-trained models did not test well.If it does not involve your academic achievements and your convenience, I would like to know how your data set is made. I just want to refer to how the data set is made and not give it to others。I really appreciate your help in the end！！！

**Comment by ZincJT** [comment]
Hello, your reply is of great help to my understanding of the paper.However, I still have some questions that need your clarification.

1. First, I have carefully derived the formula of your  reply。U_t=2U_(t-1)-U_(t-2)+t^2f=[2,-1]C_(t-1)+t^2f。It's not the same as your derivation， I don't understand why C_t here and not C_(t-1)。because C_(t-1)=[U_(t-1),U_(t-2)]。
2. Secondly,I don't understand why sf is used instead of (equation 16) in the paper。In the wave equation and in the heat conduction equation you use sf instead。Could you tell me where sf came from？
3. Thirdly,I tried to experiment with my own data set and found that the accuracy of the test was very poor. I think there is something wrong with the method of making the data set. Could you please tell me how the data set came about and the analytical solution of the equation。
4. I found that the loss functions of the two experiments are slightly different. For the source sparseness you mentioned, why is the lambd=0 of the heat conduction equation and the lambd=1e-7 of the wave equation。
I really appreciate your help,I am very sorry if the above questions involve your personal privacy。

**Comment by sahapriyabrata** [comment]
1. Yes, you are correct. It should be $C_{t-1}$ when computing $U_t$. I have updated my previous comment to show the calculation of $U_{t+1}$, instead of $U_t$. 
2. Sorry, I didn't realize that I did not upload the scripts I used for the final version of the paper. I have uploaded the modified scripts now. $sf$ is used in the new version as well. It is a scale factor used for normalization (similar to batch normalization) before feeding the data to RedNet. 
3. You probably have to modify the model if you are using your own dataset, depending on the type of system. For example, the models for heat and wave are different.  If your system has different spatial derivatives, then the model should consider those. 
We didn't solve the equations analytically because the source dynamics we used cannot be represented analytically.
4. $\lambda$ is a hyperparameter. Check fig. 7 and 9 of the paper. We compared results for different values of $\lambda$. The value of $\lambda$ in the codes corresponds to the best result, for both cases.
